### PR TITLE
Navigate to login page on app link if logged out

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: shared/actions/profile/index.js [hunk 1]
```diff
@@ -209,7 +217,7 @@ function* _profileSaga(): Saga.SagaGenerator<any, any> {
   )
   yield Saga.safeTakeEveryPure(ProfileGen.outputInstructionsActionLink, _outputInstructionsActionLink)
   yield Saga.safeTakeEveryPure(ProfileGen.showUserProfile, _showUserProfile)
-  yield Saga.safeTakeEveryPure(AppGen.link, _onAppLink)
```
Comment (chrisnojima) [inline shared/actions/profile/index.js]: this can still be pure, the second param is state so just use that and not the select

Comment (akalin-keybase) [reply shared/actions/profile/index.js]: Ah, right, fixed.

**Comment by akalin-keybase** [comment]
PTAL @chrisnojima

**Pull request merged**